ecl2df/gruptree.py

```diff
@@ -225,16 +226,24 @@ def edge_dataframe2dict(dframe):
 
     children, parents = zip(*edges)
     roots = set(parents).difference(children)
-    trees = []
-    for root in list(roots):
-        trees.append({root: subtrees[root] for root in roots})
-    return trees
+    return [{root: subtrees[root]} for root in roots]
 
 
-def dict2treelib(name, nested_dict):
-    """Convert a nested dictonary to a treelib Tree
-    object. This function is recursive.
+def _subdict2treelib(nested_dict, name):
+    """Internal recursive function for generating trees."""
+    assert isinstance(nested_dict, dict)
+    tree = treelib.Tree()
+    tree.create_node(name, name)
+    child_names = list(nested_dict.keys())
+    child_names.sort()
+    for child_name in child_names:
+        # The paste() function requires unique node names.
+        tree.paste(name, _subdict2treelib(nested_dict[child_name], child_name))
+    return tree
 
+
+def dict2treelib(nested_dict, deprecated=None):
```

Comment:
A bit surprised that `treelib` didn't have a `treelib.from_dict` or similar (or just accepted a dict on construct), as it is such a general operation. But I see on their github page that several people have commented this, and although there's some work ongoing, they've yet to finalise it. 

I'm not sure if the best way to remove the first positional argument is to add another keyword argument (that requires some context in order to understand it's existence. It's really a private keyword argument to handle the case `dict2treelib("no longer possible", {})`and forward to `dict2treelib({})`). And if someone calls `dict2treelib(name="no longer possible", nested_dict={})` it'll also raise an error. Just not sure if it is worth it.. I don't have a good suggestion yet though.

Does it need to be a nested_dict by the way? Could it not just simpy be single entry dict? Would be handled I guess?

Reply:
I intend to remove this deprecated feature, but the backwards compatibility was added as the `rft.py` file actually used it, and then possibly other scripts. Correct that all-named arguments will now fail.